src/platform/plugins/shared/dashboard/public/dashboard_api/performance/query_performance_tracking.ts

```diff
@@ -124,6 +131,41 @@ function reportPerformanceMetrics({
   const duration =
     loadType === 'dashboardSubsequentLoad' ? timeToData : Math.max(timeToData, totalLoadTime);
 
+  const groupedPerformanceMarkers = getPerformanceTrackersGroupedById('Lens');
+
+  const measurements = Object.entries(groupedPerformanceMarkers).map(([id, group]) => {
+    const markerName = group[0].name.split(':').slice(0, -1).join(':');
```

Comment:
even if I trust the logic of `groupby` because it works by construction we probably want to be sure that `group[0]` is defined

Reply:
added a check in 9e8d87f3e72.